src/audio/host.c

```diff
@@ -359,13 +359,17 @@ static uint32_t host_get_copy_bytes_normal(struct comp_dev *dev)
 		/* limit bytes per copy to one period for the whole pipeline
 		 * in order to avoid high load spike
 		 */
-		const uint32_t free_bytes =
-			audio_stream_get_free_bytes(&hd->local_buffer->stream);
```

Comment:
oh, this declaration shadowed the variable at the higher level. This was completely broken. ARGHHH.